.github/scripts/extract_git_ref.py

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env python3
+
+import os
+
+def get_branch():
+    gh_event = os.environ["GITHUB_EVENT_NAME"]
+    gh_ref = os.environ["GITHUB_REF"]
+    gh_head_ref = os.environ.get("GITHUB_HEAD_REF")  # only exists in PRs
+
+    if gh_event == "pull_request":
+        if gh_head_ref is not None:
+            msg = f"BRANCH={gh_head_ref}"
+        else:
+            raise ValueError("GITHUB_HEAD_REF isn't set")
+    elif gh_event == "push":
+        # Handle differences between branches/tags
+        branch = "refs/heads/"
+        tag = "refs/tags/"
+        if gh_ref.startswith(branch):
+            msg = f"BRANCH={gh_ref[len(branch):]}"
+        elif gh_ref.startswith(tag):
+            msg = f"BRANCH={gh_ref[len(tag):]}"
+        else:
+            raise ValueError("Unanticipated git ref from a GitHub push event")
+    elif gh_event == "workflow_dispatch":
+        # Workflow Dispatch should only be able to trigger from branches
+        branch = "refs/heads/"
+        if gh_ref.startswith(branch):
+            msg = f"BRANCH={gh_ref[len(branch):]}"
+        else:
+            raise ValueError("Unanticipated git ref from a GitHub workflow_dispatch event")
+    else:
+        raise ValueError("Unanticipated GitHub Event")
+    return msg
```

Comment:
Alright, so this is only a suggestion, you don't have to accept this change. (If you do, you might want to double-check the correctness, since I was typing this into the GH review UI, not a proper text editor).

Stylistically I tend to favor a guard clause that checks for conditions we _don't_ want and then raises an exception when it finds them, rather than having nested if-blocks that check for conditions we _do_ want. I find that it keeps indentation to a minimum, and is also a little easier to trace program execution since the blocks inside the conditions are pretty short.

That said, there was nothing wrong with your solution, so if you ignore this, my feelings will not be hurt.

```suggestion
    # Only run for pull requests, push events, and manual workflow dispatches
    if gh_event not in ["pull_request", "push", "workflow_dispatch"]:
        raise ValueError(f"Unanticipated GitHub Event: {gh_event}")

    if gh_event == "pull_request":
        if gh_head_ref is None:
            raise ValueError("GITHUB_HEAD_REF isn't set")
        return f"BRANCH={gh_head_ref}"

    # Extract the branch name from the current ref
    ref_match = re.match(r'refs/(?:heads|tags)/(.+)$', gh_ref)
    
    if not ref_match:
        raise ValueError(
            f"Unanticipated git ref from a GitHub {gh_event} event: {gh_ref}"
        )
        
    return f"BRANCH={ref_match[1]}"
```

Reply:
Thanks for the suggestion! I agree that this makes it easier to read, so thanks for suggesting it! My previous version was pretty influenced by the bash script.